app/bundles/CoreBundle/Model/FormModel.php

```diff
@@ -274,7 +275,7 @@ public function setTimestamps(&$entity, $isNew, $unlock = true): void
 
     private function setModifiedData(object $entity): void
     {
-        if (method_exists($entity, 'setDateModified') && method_exists($entity, 'getDateModified') && !$entity->getDateModified()) {
+        if (method_exists($entity, 'setDateModified') && method_exists($entity, 'getId') && $entity->getId()) {
```

Comment:
```suggestion
        if (method_exists($entity, 'setDateModified')) {
```
Can we make it simpler? I'm not sure why I added the check for `getDateModified` in https://github.com/mautic/mautic/pull/14433. The goal is to set the date modified every time no matter if the entity is being created or updated.

Reply:
Not really. When we set the modified date for a new entity, some [tests are failing](https://github.com/mautic/mautic/actions/runs/18758631431/job/53516996380?pr=15591). Therefore, I have decided to set the modified date for updated entities only by including this condition:

```php
&& method_exists($entity, 'getId') && $entity->getId()
```

Other than that, it’s a good idea to have a `null` value for a newly created entity that hasn’t been changed yet. Once it’s been modified, we set that date, and we know the entity is no longer new.

Reply:
The intent was that the date modified will be filled every time. See the use case in https://github.com/mautic/mautic/pull/14433